crates/exex/exex/src/manager.rs

```diff
@@ -618,6 +634,16 @@ impl Future for ExExManager {
     type Output = eyre::Result<()>;
 
     fn poll(mut self: Pin<&mut Self>, cx: &mut Context<'_>) -> Poll<Self::Output> {
+        // Drain the finalized header stream and grab the last finalized header
+        let mut last_finalized_header = None;
+        while let Poll::Ready(finalized_header) = self.finalized_header_stream.poll_next_unpin(cx) {
+            last_finalized_header = finalized_header;
+        }
+        // If there is a finalized header, finalize the WAL with it
+        if let Some(header) = last_finalized_header {
+            self.wal.finalize((header.number, header.hash()).into())?;
```

Comment:
this should be fine because we're operating on the assumption that exex will never trail the finalized block, right?

Reply:
wdym trail? If you're worried about an ExEx being behind the finalized block, then it's fine because we have this data in the database anyway to backfill the ExEx in case it's behind.